cli/src/commands/codegen.rs

```diff
@@ -102,13 +117,31 @@ fn codegen(
     let mut derives = DerivesRegistry::new(&crate_path);
     derives.extend_for_all(p.into_iter());
 
-    for (ty, derive) in derives_for_type.into_iter() {
+    for (ty, derive) in derives_for_type {
         let ty = syn::parse_str(&ty)?;
         let derive = syn::parse_str(&derive)?;
-        derives.extend_for_type(ty, std::iter::once(derive), &crate_path)
+        derives.extend_for_type(ty, std::iter::once(derive), &crate_path);
     }
 
-    let type_substitutes = TypeSubstitutes::new(&crate_path);
+    let mut type_substitutes = TypeSubstitutes::new(&crate_path);
+    for (from_str, to_str) in substitute_types {
+        let from: syn::Path = syn::parse_str(&from_str)?;
+        let to: syn::Path = syn::parse_str(&to_str)?;
+        let to = to.try_into().map_err(|e: TypeSubstitutionError| {
+            eyre::eyre!(
+                "Cannot parse substitute '{from_str}={to_str}': {}",
+                e.to_string()
+            )
+        })?;
+        type_substitutes
+            .insert(from, to)
+            .map_err(|e: TypeSubstitutionError| {
+                eyre::eyre!(
+                    "Cannot parse substitute '{from_str}={to_str}': {}",
+                    e.to_string()
```

Comment:
`TypeSubstitionError` already implements `Display` right and the extra `to_string` can be removed?